src/bogart/AS_BAT_SplitTig_ByRegions.C

```diff
@@ -0,0 +1,120 @@
+
+/******************************************************************************
+ *
+ *  This file is part of canu, a software program that assembles whole-genome
+ *  sequencing reads into contigs.
+ *
+ *  This software is based on:
+ *    'Celera Assembler' r4587 (http://wgs-assembler.sourceforge.net)
+ *    the 'kmer package' r1994 (http://kmer.sourceforge.net)
+ *
+ *  Except as indicated otherwise, this is a 'United States Government Work',
+ *  and is released in the public domain.
+ *
+ *  File 'README.licenses' in the root directory of this distribution
+ *  contains full conditions and disclaimers.
+ */
+
+#include "AS_BAT_SplitTig.H"
+
+
+uint32
+splitTigByRegions(TigVector                &tigs,
+                  Unitig                   *tig,
+                  vector<breakPointCoords> &BP,
+                  Unitig                  **newTigs,
+                  int32                    *lowCoord,
```

Comment:
A bit weird to see so many arrays of the same size. Any chance it could make sense to use single array of a struct that describes the result? And maybe even return that vector?